From b283724419b219cfe5941b250c0f9c17b721e3e7 Mon Sep 17 00:00:00 2001
From: Hyun Kwon <hyun.kwon@xilinx.com>
Date: Tue, 11 Aug 2020 18:38:40 -0700
Subject: [PATCH 1510/1852] misc: xilinx-ai-engine: use correct printk formats
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Simple printk format corrections:
- %zx for size_t
- %pa for phys_addr_t,

drivers/misc/xilinx-ai-engine/ai-engine-dma.c:83:24: warning: format ‘%lx’ expects argument of type ‘long unsigned int’, but argument 4 has type ‘size_t {aka unsigned int}’ [-Wformat=]
   dev_err(&apart->dev, "failed to find vma for %p, 0x%lx.\n",
                      ^

drivers/misc/xilinx-ai-engine/ai-engine-dma.c:103:4: warning: format ‘%lx’ expects argument of type ‘long unsigned int’, but argument 4 has type ‘size_t {aka unsigned int}’ [-Wformat=]
    "failed to get dma address for %p, 0x%lx.\n", va, len);
    ^

drivers/misc/xilinx-ai-engine/ai-engine-mem.c:109:6: warning: format ‘%llx’ expects argument of type ‘long long unsigned int’, but argument 5 has type ‘phys_addr_t {aka unsigned int}’ [-Wformat=]
      "failed to mmap (%u,%u)memory, remap failed, 0x%llx, 0x%lx.\n",
      ^
Reported-by: kernel test robot <lkp@intel.com>
Signed-off-by: Hyun Kwon <hyun.kwon@xilinx.com>
Reviewed-by: Wendy Liang <wendy.liang@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/misc/xilinx-ai-engine/ai-engine-dma.c | 4 ++--
 drivers/misc/xilinx-ai-engine/ai-engine-mem.c | 4 ++--
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/misc/xilinx-ai-engine/ai-engine-dma.c b/drivers/misc/xilinx-ai-engine/ai-engine-dma.c
index e382ba62d6fd..641bb62a413c 100644
--- a/drivers/misc/xilinx-ai-engine/ai-engine-dma.c
+++ b/drivers/misc/xilinx-ai-engine/ai-engine-dma.c
@@ -80,7 +80,7 @@ static dma_addr_t aie_part_get_dmabuf_da(struct aie_partition *apart,
 
 	vma = find_vma(current->mm, va_start);
 	if (!vma) {
-		dev_err(&apart->dev, "failed to find vma for %p, 0x%lx.\n",
+		dev_err(&apart->dev, "failed to find vma for %p, 0x%zx.\n",
 			va, len);
 		return 0;
 	}
@@ -100,7 +100,7 @@ static dma_addr_t aie_part_get_dmabuf_da(struct aie_partition *apart,
 	 */
 	if (va_off + len >= adbuf->attach->dmabuf->size) {
 		dev_err(&apart->dev,
-			"failed to get dma address for %p, 0x%lx.\n", va, len);
+			"failed to get dma address for %p, 0x%zx.\n", va, len);
 		return 0;
 	}
 
diff --git a/drivers/misc/xilinx-ai-engine/ai-engine-mem.c b/drivers/misc/xilinx-ai-engine/ai-engine-mem.c
index 776112a32a91..b4fb217492b1 100644
--- a/drivers/misc/xilinx-ai-engine/ai-engine-mem.c
+++ b/drivers/misc/xilinx-ai-engine/ai-engine-mem.c
@@ -106,8 +106,8 @@ static int aie_mem_mmap(struct dma_buf *dmabuf, struct vm_area_struct *vma)
 					      len, vma->vm_page_prot);
 			if (ret) {
 				dev_err(&apart->dev,
-					"failed to mmap (%u,%u)memory, remap failed, 0x%llx, 0x%lx.\n",
-					loc.col, loc.row, mempa, len);
+					"failed to mmap (%u,%u)memory, remap failed, 0x%pa, 0x%lx.\n",
+					loc.col, loc.row, &mempa, len);
 				return ret;
 			}
 			addr += len;
-- 
2.31.1

